src/Markdig/Helpers/HtmlHelper.cs

```diff
@@ -617,7 +617,7 @@ public static int ScanEntity(string s, int pos, int length, out string namedEnti
             {
                 // expect a letter and 1-31 letters or digits
                 c = s[pos + 1];
-                if ((c < 'A' || c > 'Z') && (c < 'a' && c > 'z'))
+                if (!((c >= '0' && c <= '9') || (c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z')))
```

Comment:
yes but not adding `(c >= '0' && c <= '9')` (comment is quite troubling: It is a letter, followed by letters or digits